bridges/PixivBridge.php

```diff
@@ -1,72 +1,109 @@
 <?php
 class PixivBridge extends BridgeAbstract {
 
-	const MAINTAINER = 'teromene';
+	const MAINTAINER = 'Yaman Qalieh';
 	const NAME = 'Pixiv Bridge';
 	const URI = 'https://www.pixiv.net/';
 	const DESCRIPTION = 'Returns the tag search from pixiv.net';
+	const CACHE_TIMEOUT = 21600; // 6h
 
 
 	const PARAMETERS = array( array(
+		'mode' => array(
+			'name' => 'Post Type',
+			'type' => 'list',
+			'values' => array('Illustration' => 'illustrations/',
+							  'Manga' => 'manga/',
+							  'Novel' => 'novels/')
+		),
 		'tag' => array(
-			'name' => 'Tag to search',
-			'exampleValue' => 'example',
+			'name' => 'Query to search',
+			'exampleValue' => '葬送のフリーレン',
 			'required' => true
 		),
+		'posts' => array(
+			'name' => 'Post Limit',
+			'type' => 'number',
+			'defaultValue' => '10'
+		),
+		'fullsize' => array(
+			'name' => 'Full-size Image',
+			'type' => 'checkbox'
+		)
 	));
 
-	public function collectData(){
-
-		$html = getContents(static::URI . 'search.php?word=' . urlencode($this->getInput('tag')))
-			or returnClientError('Unable to query pixiv.net');
-		$regex = '/<input type="hidden"id="js-mount-point-search-result-list"data-items="([^"]*)/';
-		$timeRegex = '/img\/([0-9]{4})\/([0-9]{2})\/([0-9]{2})\/([0-9]{2})\/([0-9]{2})\/([0-9]{2})\//';
-
-		preg_match_all($regex, $html, $matches, PREG_SET_ORDER, 0);
-		if(!$matches) return;
-
-		$content = json_decode(html_entity_decode($matches[0][1]), true);
+	const JSON_KEY_MAP = array(
+		'illustrations/' => 'illust',
+		'manga/' => 'manga',
+		'novels/' => 'novel'
+	);
+	const WORK_LINK_MAP = array(
+		'illustrations/' => 'artworks/',
+		'manga/' => 'artworks/',
+		'novels/' => 'novel/show.php?id='
+	);
+
+	public function collectData() {
+		$content = getContents($this->getSearchURI())
+				 or returnClientError('Unable to query pixiv.net');
```

Comment:
getContents raises exception on non 200 http result. No need to use returnClientError.